### ISSUE TITLE
ELAN fingerprint drivers

### ISSUE THREAD

**Issue by 1457384613gh** [sebanc/brunch#1744]
ELAN fingerprint drivers.
Do they work?

**Comment by imatefx** [comment]
I just installed chromeOS on my Asus Flow X13 its ELAN fingerprint is not detected, but it shows up in Manage usb devices in Linux development settings

**Comment by tejasraman** [comment]
ChromeOS has _NO_ Elan fingerprint support whatsoever @1457384613gh @imatefx

**Comment by tejasraman** [comment]
Never really experimented with lvm, sorry :)

**Comment by tejasraman** [comment]
@1457384613gh I do, but LVM was never really a practical solution for me.

**Comment by tejasraman** [comment]
I think it can if you have windows installed as mbr.

**Comment by tejasraman** [comment]
Ohhhhhhhh... No, it won't do that.
You might be able to do this using clover but that's the extent of my knowledge on the situation

**Comment by tejasraman** [comment]
Refind isn't clover.....
You need the clover loader to load an efi file for you

**Comment by tejasraman** [comment]
Ok so duet loader(from opencore I assume) worked :)

**Comment by tejasraman** [comment]
What about Brunch's mbr patch?

Also, Duet seems like a good way to emulate EFI on BIOS systems. Thanks, I never really thought of duet as a solution for anything other than opencore (and I don't have any old bios systems anymore, my oldest is a 2011 Mac and a 2013 (8.1-era) PC)

**Comment by tejasraman** [comment]
Ok so your laptop has no efi support?

And I do understand that the mbr patch has no multi boot support……

**Comment by tejasraman** [comment]
1st gen hasn’t been supported since ChromeOS 81.

**Comment by tejasraman** [comment]
![image](https://user-images.githubusercontent.com/96961804/226122280-1eaff528-71d4-490d-b071-f10140fbf2cf.jpeg)

**Comment by tejasraman** [comment]
Sorry, I'm confused. M450 and 45xx series are 2 different things :)

**Comment by tejasraman** [comment]
And there is a GPU ROM patch to add UEFT Graphics Output Protocol (GOP) support to GPUs, but it's only for dedicated NVidia/AMD cards.

**Comment by tejasraman** [comment]
Have you tried logging in to google? It will fail, I’ve tried it with 2nd gen (and it had a UEFi GPU because of Apple too) and it fails.

**Comment by tejasraman** [comment]
If you check a `dmesg` after logging in you’ll see a lot of “invalid CPU instruction” messages.

reven doesn’t rely on those ISAs but it has no Play Store either.
I guess fyde is your best option :)

**Comment by tejasraman** [comment]
ChromeOS (Settings) is not part of the grub; rather, it can even be run from inside of chromeOS `sudo edit-brunch-config`


And btw, I'm not the Brunch developer, @sebanc is.

**Comment by tejasraman** [comment]
Ok yes I am :)
I'll look into your proposals but "contributor" means I've submitted a few PRs :)
I'm making a grub theme to test for future implementations, but you need to ask someone else about ventoy :). (I love ventoy but I know nothing about it beyond the basics despite 5out of 6 of my boot USBs using it)

**Comment by tejasraman** [comment]
And the bootsplashes are all in .png format and at static _960x540_ (shocking, I know) and are surrounded by (black) borders on higher resolution displays (e.g 1920x1080 - try a light theme and see for yourself). 

I can see the value of vector bootsplashes that are truly scalable (try booting brunch in 4K and it'll be so small) but I would need to look into it more.

Also, a progress bar for the bootsplashes certainly isn't impossible (at least for rootFS where there is a percentage in verbose mode)

**Comment by tejasraman** [comment]
Ok here's what I kno
![2023-03-18_22-59](https://user-images.githubusercontent.com/96961804/226154673-195ccf03-6435-48fe-970c-e017f96ab148.png)
- Brunch is using FBV to display generic images on screen.
- Brunch is manually appending logs to kmsg

**Comment by tejasraman** [comment]
> Could FBV support to show .SVG?
No, FBV is old and cannot do that. At this point an alternative needs to be found if you want to have SVG boot splashes.

**Comment by tejasraman** [comment]
Ok thanks @1457384613gh I didn't know that :)

But the exciting part is that there is ext4 support (and brunch performs best with ext4, not NTFS)

**Comment by tejasraman** [comment]
@1457384613gh I know that Ventoy boots internal grub. I'm not sure if brunch images have an internal ESP.

**Comment by tejasraman** [comment]
Ok thanks so much @1457384613gh

I'll test some things and try to make it work as best I can..

**Comment by tejasraman** [comment]
Because I'm not "in contact" with every Brunch contributor; contributions are usually done individually.

**Comment by tejasraman** [comment]
And I’m not the person to ask about Ventoy either… I’ll try though

**Comment by tejasraman** [comment]
@1457384613gh For vectorized (non-PNG) bootsplashes I learned about [jfbview](https://github.com/jichu4n/jfbview), a PDF viewer for framebuffers. This will allow high resolution display of bootsplashes

**Comment by tejasraman** [comment]
Yes, my ID for element/matrix is traman124:matrix.org

**Comment by tejasraman** [comment]
That looks like a verbatim copy of brunch’s grub2…..

**Comment by tejasraman** [comment]
Also, there is no way to use ttfs and pngs (like in Plymouth) with a simple ttf tool. The closest that I could get was jfbview (which shows hi res bootsplashes) (see my fork of brunch-unstable)

**Comment by tejasraman** [comment]
I see no point in including a txt file system in brunch; people aren't supposed to edit grub.cfg. I don't foresee a large number of people using non stock locations for the img files...

Can you explain the potential usage for this???
